bindings/python/src/session.cpp

```diff
@@ -206,68 +206,123 @@ namespace
+                // without holding the GIL and likely crash.
+                // https://mail.python.org/pipermail/cplusplus-sig/2007-June/012130.html
+                p.ti = std::make_shared<torrent_info>(
+                    extract<torrent_info const&>(params[key]));
+                continue;
+            }
+            else if(key == "info_hash")
+            {
+                p.info_hash = sha1_hash(
+                bytes(extract<bytes>(params[key])).arr.data());
+                continue;
+            }
+            else if(key == "name")
+            {
+                p.name = extract<std::string>(params[key]);
+                continue;
+            }
+            else if(key == "save_path")
+            {
+                p.save_path = extract<std::string>(params[key]);
+                continue;
+            }
 #ifndef TORRENT_NO_DEPRECATE
-        if (params.has_key("uuid"))
-            p.uuid = extract<std::string>(params["uuid"]);
+            else if(key == "resume_data")
+            {
+                std::string resume = extract<std::string>(params[key]);
+                p.resume_data.assign(resume.begin(), resume.end());
+                continue;
+            }
+            else if(key == "uuid")
+            {
+                p.uuid = extract<std::string>(params["uuid"]);
+                continue;
+            }
 #endif
-
-        if (params.has_key("file_priorities"))
-        {
-            list l = extract<list>(params["file_priorities"]);
-            int const n = int(boost::python::len(l));
-            for(int i = 0; i < n; i++)
-                p.file_priorities.push_back(extract<std::uint8_t>(l[i]));
-            p.file_priorities.clear();
+            else if(key == "storage_mode")
+            {
+                p.storage_mode = extract<storage_mode_t>(params[key]);
+                continue;
+            }
+            else if(key == "trackers")
+            {
+                p.trackers = extract<std::vector<std::string>>(params[key]);
+                continue;
+            }
+            else if(key == "url_seeds")
+            {
+                p.url_seeds = extract<std::vector<std::string>>(params[key]);
+                continue;
+            }
+            else if(key == "http_seeds")
+            {
+                // TODO: is it possible to create map["http_seeds"]=add_torrent_params::http_seeds?
+                // I don't how how to do it in c++ because we don't have reflections
+                // but I would like a map instead of if else and decltype does help us
+                p.http_seeds =
+                    extract<decltype(add_torrent_params::http_seeds)>(params[key]);
+                continue;
+            }
+            else if(key == "dht_nodes")
+            {
+                // TODO: did not find converter
+                //p.dht_nodes =
+                //    extract<std::vector<std::pair<std::string, int>>>(params[key]);
```

Comment:
I think it would be better to add the converter to bindings/python/src/converters.cpp
It should be straight-forward of how to do that. just follow the example of the other ones